### ISSUE TITLE
Auto Levels: calculate based on cropping

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#709]
Originally reported on Google Code with ID 720

```
Currently auto levels are calculated based on the full image area.
Improvement suggestion: calculate based on image content within the crop.

Ideally, there should be a checkbox next to Auto levels button- "crop-based".
When crop is changed and auto levels are selected and "crop-based" checkbox is selected
auto levels should be recalculated.

This feature would allow to perform exposure setting based on selected region of interest.
User may temporarily set cropping to the region of interest, execute auto levels "measurement",
deselect auto levels button (EC and black stay unchanged (this is current functionality)),
set final crop.
```

Reported by `michaelezra000` on 2011-06-06 01:59:37

**Comment by leodp** [comment]
Hi, I would like to add an user case: Old slides digitized with a DSLR
1) The edge of the acquired image is directly illuminated by the slides light (maybe passing through the film holes)=>This defines the white level
2) The edge of the film is darker than the black level of the slides image=>This defines the black level
1 & 2 above cause wrong auto exposure correction and also wrong white balance correction.
The issue can be overcome if it would be possible to define a crop area (center of the image) for the auto exposition and AWB (similarly to what already done by spot WB, but more generic). The .pp3 profile with the specific crop area can then be batch-applied to all the scanned images.
Baseline issue: scanning hundreds of slides is quite time consuming. RAW computerized processing (read:rawtherapee) should not take more time than the scanning itself, but profit from the automatic nature of information processing.
Searching with google "digitizing slides" returns 337000 results. This open issue may be of interest to more than only a few "lazy persons". 
https://encrypted.google.com/search?q=digitizing+slides&hl=en&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjPpZ783L7WAhUKvBoKHWSdCt0Q_AUIDCgD&biw=1032&bih=543

![imgp8002](https://user-images.githubusercontent.com/17860979/30786909-2cb7ffa4-a17e-11e7-8068-468541506eb8.jpg)
Leo

**Comment by heckflosse** [comment]
@leodp There are thoughts to allow a raw crop in rt. That would mean a crop of the raw file directly after decoding phase. That would solve your issue. But currently these are just thoughts...

**Comment by leodp** [comment]
That would help. But will sacrifice the film and the image edges, which are part of the flair these images have.
I like particularly the not straight edge between the image and the black frame of the film.
But this is just my thought.

**Comment by heckflosse** [comment]
@leodp I see your point, but that's a different issue then. This one is about `Auto Levels: calculate based on cropping`

**Comment by nahuelaguilar93** [comment]
I agree that this would be a great feature. Given the possition of my white balance card in the image, this will allow me to automatically crop and auto white balance the image by just properly generating a pp3. Hope it's available soon.

**Comment by deusexmamiya** [comment]
Another vote for this. It's REALLY needed for timelapses under changing light conditions.

Here's an example. Two consecutive frames from a Milky Way timelapse. When the second camera's LCD screen changed in the bottom corner to a fainter, less blue light, the AutoWB changed for the whole image, and made the yellowish Milky Way less visible. If I could **user-define an "Auto WB box"**, covering in this example the upper 80% of all the images, it would avoid this area where the light changes. 

![RT_timelapse_AutoWB_changes](https://user-images.githubusercontent.com/68204605/127148208-886dc39e-bc5d-49a7-9cb5-029fdc99c221.jpg)

Now if you're wondering "_Why doesn't he set a custom WB and use it for all the timelapse frames?_" - that's even worse. The sky was entering twilight (becoming gradually much bluer), so the Milky Way stands out much better when Auto WB cancels out this progressive blue-ing. A fixed WB just fails completely. Auto WB is best, but it really needs the addition of that extra control over what block of the image is included in determining Auto WB. 
(Note - this is separate to the cropping box. I don't want to crop the images at all in this case).

**Comment by nomar500** [comment]
I thought I had already voted for this feature.
👍

**Comment by Olivert0187** [comment]
Has there been any progress on this? I have been googling and looking through the code but can't find anything. Is it possible to add so that auto balancing functions are only based on what is inside the crop? Would it be possible to implement?

**Comment by rotanov** [comment]
+vote
